src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -179,6 +180,31 @@ def sort_code_mapping_configs(
                     sorted_configs.insert(0, config)
         return sorted_configs
 
+    def get_commit_sha(
+        self,
+        integration: Integration,
+        organization_id: str,
+        line_no: int,
+        filepath: str,
+        current_config: JSONData,
+    ) -> Optional[str]:
+        commitSha = None
+        integrationInstallation = integration.get_installation(organization_id=organization_id)
+        try:
+            git_blame_list = integrationInstallation.get_blame_for_file(
+                current_config["repository"],
+                filepath,
+                current_config["config"]["defaultBranch"],
+                line_no,
+            )
+            for blame in git_blame_list:
+                if line_no >= blame["startingLine"] and line_no <= blame["endingLine"]:
+                    commitSha = str(blame["commit"]["oid"])
+                    break
+        except Exception:
+            logger.exception("Could not get git blame")
```

Comment:
We need a test case for this:
<img width="494" alt="image" src="https://user-images.githubusercontent.com/44410/212391229-edebf7b5-4873-4f62-94cd-4b8e030ecb40.png">

Feel free to follow up on another PR since this is almost there.